src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -60659,5 +60681,1008 @@ public void TestMeetsAgree()
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(T)x").WithLocation(5, 13),
+                // (8,13): warning CS8629: Nullable value type may be null.
+                //         _ = (T)x; // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "(T)x").WithLocation(8, 13));
+        }
+
+        [Fact]
+        public void NullableT_08()
+        {
+            var source =
+@"class Program
+{
+    static void F1((int, int) x)
+    {
+        (int, int)? y = x;
+        _ = y.Value;
+        var z = ((int, int))y;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NullableT_09()
+        {
+            var source =
+@"class Program
+{
+    static void F1((int, int)? x)
+    {
+        var y = ((int, int))x; // 1
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (5,17): warning CS8629: Nullable value type may be null.
+                //         var y = ((int, int))x; // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "((int, int))x").WithLocation(5, 17));
+        }
+
+        [Fact]
+        public void NullableT_10()
+        {
+            var source =
+@"class Program
+{
+    static void F<T>(T t) where T : struct
+    {
+        T? x = null;
+        _ = x.Value; // 1
+        T? y = default;
+        _ = y.Value; // 2
+        T? z = default(T);
+        _ = z.Value;
+        T? w = t;
+        _ = w.Value;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (6,13): warning CS8629: Nullable value type may be null.
+                //         _ = x.Value; // 1
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "x").WithLocation(6, 13),
+                // (8,13): warning CS8629: Nullable value type may be null.
+                //         _ = y.Value; // 2
+                Diagnostic(ErrorCode.WRN_NullableValueTypeMayBeNull, "y").WithLocation(8, 13));
+        }
+
+        [WorkItem(31502, "https://github.com/dotnet/roslyn/issues/31502")]
+        [Fact]
+        public void NullableT_11()
+        {
+            var source =
+@"class Program
+{
+    static void F<T>(T t) where T : struct
+    {
+        T? x = new System.Nullable<T>();
+        _ = x.Value; // 1
+        T? y = new System.Nullable<T>(t);
+        _ = y.Value;
+        T? z = new T?();
+        _ = z.Value; // 2
+        T? w = new T?(t);
+        _ = w.Value;
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            // https://github.com/dotnet/roslyn/issues/31502: Recognize Nullable<T> constructors.
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void NullableT_12()
```

Comment:
Is this substantially different from `NullableT_06`, other than also having the negative case? If not, consider consolidating.